Bug? Missing keys after backup then restore of a large db

I'm having trouble grasping the results of backing up and restoring a badgered database.
I have a database which has a couple thousand entries. I run badger_info which gives me the result below:
```
> badger_info --dir original_db
[Manifest]
MANIFEST           1562  2018-01-18T10:37:32+01:00
[Level 0]
000085.sst        15587  2018-01-18T10:22:43+01:00 0
[Level 1]
000084.sst       362235  2018-01-15T19:39:06+01:00 1
[Value Log]
000000.vlog     4286777  2018-01-18T10:37:32+01:00
[Summary]
Level 0 size: 15587
Level 1 size: 362235
Total index size: 377822
Value log size: 4286777
Abnormalities: None.
0 extra files.
0 missing files.
0 empty files.
0 truncated manifests.
```
I run badger backup to backup the the database. Then i restore the database to a different folder and run badger_info which then gives me the result below.

```
> badger_info --dir new_db
[Manifest]
MANIFEST             28  2018-01-18T11:03:15+01:00
[Level 0]
000001.sst       111648  2018-01-18T11:03:15+01:00 0
[Value Log]
000000.vlog     3414691  2018-01-18T11:03:15+01:00
[Summary]
Level 0 size: 111648
Total index size: 111648
Value log size: 3414691
Abnormalities: None.
0 extra files.
0 missing files.
0 empty files.
0 truncated manifests.
```
I assume it compacted the database to just level 0 based on this result. if i tail the value log i can see that it has at least most of the entries.

This all looks fine, based on my assumptions, until i iterate all keys using the snippet from the docs. It returns only 3 keys.
This does not seem to correlate with the metric data returned by badger_info

@janardhan1993  unfortunately the fix in master didn't fix the case below:
1) First insert some rows into DB1
2) Backup it
3) Restore the result snapshot into DB2
4) Set the same keys
5) Backup DB2
6) Restore DB2 into DB3
7) Keys are not there
The gist which replicates this:
https://gist.github.com/Draal/aeec21ee8260e33e2bf4de2bd4ecb4d4

```go test
backup1 length: 24320
backup2 length: 48800
--- FAIL: Test_BackupRestore (0.10s)
	badger_snapshot_test.go:135: Key [116 101 115 116 115 110 97 112 115 104 111 116 58 107 101 121 49 0] has been not found, but was set

FAIL```

--- a/backup.go
+++ b/backup.go
@@ -134,7 +134,7 @@ func (db *DB) Load(r io.Reader) error {
 			if err := batchSetAsyncIfNoErr(entries); err != nil {
 				return err
 			}
-			entries = entries[:0]
+			entries = make([]*Entry, 0, 1000)
 		}
 	}
 

--- a/backup_test.go
+++ b/backup_test.go
@@ -17,8 +17,12 @@
 package badger
 
 import (
+	"bytes"
+	"fmt"
 	"io/ioutil"
 	"os"
+	"path/filepath"
+	"reflect"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -104,4 +108,150 @@ func TestDumpLoad(t *testing.T) {
 		return nil
 	})
 	require.NoError(t, err)
+}
+
+func Test_BackupRestore(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "badger-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		os.RemoveAll(tmpdir)
+	}()
+
+	s1Path := filepath.Join(tmpdir, "test1")
+	s2Path := filepath.Join(tmpdir, "test2")
+	s3Path := filepath.Join(tmpdir, "test3")
+
+	opts := DefaultOptions
+	opts.Dir = s1Path
+	opts.ValueDir = s1Path
+	db1, err := Open(opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	key1 := []byte("key1")
+	key2 := []byte("key2")
+	rawValue := []byte("NotLongValue")
+	N := byte(251)
+	err = db1.Update(func(tx *Txn) error {
+		if err := tx.Set(key1, rawValue); err != nil {
+			return err
+		}
+		return tx.Set(key2, rawValue)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	for i := byte(0); i < N; i++ {
+		err = db1.Update(func(tx *Txn) error {
+			if err := tx.Set(append(key1, i), rawValue); err != nil {
+				return err
+			}
+			return tx.Set(append(key2, i), rawValue)
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	var backup bytes.Buffer
+	_, err = db1.Backup(&backup, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("backup1 length:", backup.Len())
+
+	opts = DefaultOptions
+	opts.Dir = s2Path
+	opts.ValueDir = s2Path
+	db2, err := Open(opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = db2.Load(&backup)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	/*
+		for i := byte(0); i < N; i++ {
+			err = db2.View(func(tx *Txn) error {
+				k := append(key1, i)
+				item, err := tx.Get(k)
+				if err != nil {
+					if err == ErrKeyNotFound {
+						return fmt.Errorf("Key %q has been not found, but was set\n", k)
+					}
+					return err
+				}
+				v, err := item.Value()
+				if err != nil {
+					return err
+				}
+				if !reflect.DeepEqual(v, rawValue) {
+					return fmt.Errorf("Values not match, got %v, expected %v", v, rawValue)
+				}
+				return nil
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	*/
+
+	for i := byte(0); i < N; i++ {
+		err = db2.Update(func(tx *Txn) error {
+			if err := tx.Set(append(key1, i), rawValue); err != nil {
+				return err
+			}
+			return tx.Set(append(key2, i), rawValue)
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	backup.Reset()
+	_, err = db2.Backup(&backup, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("backup2 length:", backup.Len())
+	opts = DefaultOptions
+	opts.Dir = s3Path
+	opts.ValueDir = s3Path
+	db3, err := Open(opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = db3.Load(&backup)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for i := byte(0); i < N; i++ {
+		err = db3.View(func(tx *Txn) error {
+			k := append(key1, i)
+			item, err := tx.Get(k)
+			if err != nil {
+				if err == ErrKeyNotFound {
+					return fmt.Errorf("Key %q has been not found, but was set\n", k)
+				}
+				return err
+			}
+			v, err := item.Value()
+			if err != nil {
+				return err
+			}
+			if !reflect.DeepEqual(v, rawValue) {
+				return fmt.Errorf("Values not match, got %v, expected %v", v, rawValue)
+			}
+			return nil
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
 }

--- a/backup_test.go
+++ b/backup_test.go
@@ -173,31 +173,29 @@ func Test_BackupRestore(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	/*
-		for i := byte(0); i < N; i++ {
-			err = db2.View(func(tx *Txn) error {
-				k := append(key1, i)
-				item, err := tx.Get(k)
-				if err != nil {
-					if err == ErrKeyNotFound {
-						return fmt.Errorf("Key %q has been not found, but was set\n", k)
-					}
-					return err
-				}
-				v, err := item.Value()
-				if err != nil {
-					return err
-				}
-				if !reflect.DeepEqual(v, rawValue) {
-					return fmt.Errorf("Values not match, got %v, expected %v", v, rawValue)
-				}
-				return nil
-			})
+	for i := byte(0); i < N; i++ {
+		err = db2.View(func(tx *Txn) error {
+			k := append(key1, i)
+			item, err := tx.Get(k)
 			if err != nil {
-				t.Fatal(err)
+				if err == ErrKeyNotFound {
+					return fmt.Errorf("Key %q has been not found, but was set\n", k)
+				}
+				return err
 			}
+			v, err := item.Value()
+			if err != nil {
+				return err
+			}
+			if !reflect.DeepEqual(v, rawValue) {
+				return fmt.Errorf("Values not match, got %v, expected %v", v, rawValue)
+			}
+			return nil
+		})
+		if err != nil {
+			t.Fatal(err)
 		}
-	*/
+	}
 
 	for i := byte(0); i < N; i++ {
 		err = db2.Update(func(tx *Txn) error {

